open-craft/sprints
# [BB-1816] Cache worklogs and lock "Complete Sprint" button

This implements caching the worklogs. Its purpose is to be as transparent as possible, so it doesn't alter the underlying logic.
This also includes locking the "Complete Sprint button" (described in the instructions), which needed caching changes to work properly.

I also updated Python to have more concise syntax with [assignment expressions (PEP 572)](https://www.python.org/dev/peps/pep-0572/).

## Testing instructions 
1. Set `.env` file (you can find it in a comment on the ticket).
1. Start the backend with `docker-compose -f local.yml up --build`.
1. Set frontend/.env.local file (you can find it in a comment on the ticket).
1. Navigate to frontend and run `npm i && npm start`.
1. Log in.
1. Open http://localhost:5555/tasks (you can find the credentials in `.env` file) and wait for the tasks to complete. This should take around 5 minutes.
1. Open browser's console and log into the dashboard. You should see that `?year=*` and `?from=*` requests are now taking under 400ms instead of over 30s.
1. Go to the cell's dashboard and press `Complete Sprint` button. Refresh the page and hover over the disabled button to see that the reason is displayed properly there.
